### DISCUSSION TITLE
Eshell history to include bash_history

### DISCUSSION THREAD

**Discussion by ratnesh1729** [minad/consult#1043] [Q&A] [answered]
When I press consult-history I see all eshell history nicely. 

I would like to have this history to include bash_history = how can I achieve this?

I came across this snippet from ivy - and I replaced ivy-read with `consult--read`. But now it shows all lines in one go. And as seen it wont include eshell history.

```
(defun me/bash-history ()
  (interactive)
  (let ((command (with-temp-buffer
                   (insert-file-contents-literally "~/.bash_history")
                   (let ((history-list (split-string (buffer-string) "\n" t)))
                     (consult--read "Command: " history-list :fuzzy t)))))
    (when command
      (insert command))))
```

Looking forward to pointers to try out.

**Comment by minad** [comment, marked as answer]
`consult-history` accesses history variables, depending on context. Accessing the bash history is out of scope for Consult.

> I came across this snippet from ivy - and I replaced ivy-read with consult--read. But now it shows all lines in one go. And as seen it wont include eshell history.

I suggest to use `completing-read` instead of `consult--read` here. In any case, make sure to use the arguments correctly. Read the manual and doc strings regarding that. Simply replacing `ivy-read` with `consult--read` won't fly.
